controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -13,6 +13,10 @@ const InitDiscoveryTopicData = async (req, res) => {
   const userId = req.userId;
 
   try {
+    let totalDataQuery = `SELECT 
+                              count(A.topic_id) as total_data
+                          FROM topics A`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query `totalDataQuery` is currently counting all topics in the database. If there are any conditions or filters applied to the `suggestedTopicsQuery`, they should also be applied to `totalDataQuery` to ensure that the total count of data is accurate and consistent with the data being returned.

```diff
-    let totalDataQuery = `SELECT 
-                              count(A.topic_id) as total_data
-                          FROM topics A`;
+    let totalDataQuery = `SELECT 
+                              count(A.topic_id) as total_data
+                          FROM topics A
+                          WHERE <your_conditions_here>`;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->